### PR TITLE
[example] add example for recognizer with grammar

### PR DESCRIPTION
**Pull request by kawaki-san**
This provides a quick demo of the using `Recognizer::new_with_grammar` API

### CODE REVIEW

**Review by Bear-03 — CHANGES_REQUESTED** [review]
Thanks for the contribution!

File: vosk/examples/grammar.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        // Anything else will be returned as [unk]
+        &[
+            "oh one two three four five six seven eight nine zero",
+            "[unk]",
+        ],
+    )
+    .expect("Could not create the recognizer");
+
+    for sample in samples.chunks(4000) {
+        let stream = recognizer.accept_waveform(sample);
```
[the remark this answers is not in the document]
Comment (kawaki-san) [reply vosk/examples/grammar.rs:40]: How about just `state` as per [mic example](https://github.com/Bear-03/vosk-rs/blob/1e7db5a1f09808b8003541830a4e44d986c154f4/vosk/examples/microphone.rs#L97) or alternatively, matching directly on `Recognizer::accept_waveform` for less verbosity

[the remark this answers is not in the document]
Comment (Bear-03) [reply vosk/examples/grammar.rs:40]: Both options seem good, but matching directly would make the line too long, so I prefer `state`, which also makes things consistent. Thanks for pointing it out!

[the remark this answers is not in the document]
Comment (kawaki-san) [reply vosk/examples/grammar.rs]: Sorted!

**Review by Bear-03 — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**